Source/Particles/ParticleCreation/FilterCopyTransform.H

```diff
@@ -72,8 +74,9 @@ Index filterCopyTransformParticles (DstTile& dst, SrcTile& src, Index* mask, Ind
     {
         if (mask[i])
         {
-            copy(dst_data, src_data, i, p_offsets[i] + dst_index);
-            transform(dst_data, src_data, i, p_offsets[i] + dst_index);
+            for (int j = 0; j < N; ++j)
+                copy(dst_data, src_data, i, N*p_offsets[i] + dst_index + j);
+            transform(dst_data, src_data, i, N*p_offsets[i] + dst_index);
```

Comment:
I think within the `transform` function it would be useful to have access to `N`. When splitting particles in 3D, we may want to create 6, 7, 8, 9 or 12 particles for instance, this being decided at runtime (right now we only have the option to create 6 or 8 particles in 3D).

I can think of two ways to do this:
- leave this code exactly as is, and template `SplittingTransformFunc` over the number of split particles,
- add an input argument to the `transform` functions.

Do you have any other way, or do you have any preference?

Reply:
The way I was picturing this working was that each elementary process would have a class associated with it that knows how to produce instances of the appropriate `filter` and `transform` functors, and this class would know the multiplicity as member data. Then, we wouldn't need to pass the number in to the `transform` functor. 

Originally, I implemented the `multiplicity` as a function argument, then changed it to a template parameter. If the multiplicity isn't known at compile time, maybe it is best to go back to the function argument approach.

Reply:
Would be interested to hear if anyone has opinions on the best interface here...

Reply:
Let's keep it this way for now. so far, we don't have that many options for splitting, we'll see if it needs to be changed later.